CMakeLists.txt

```diff
@@ -296,4 +316,414 @@ if(CMAKE_HOST_UNIX)
+	target_include_directories(GpFontHandler_FreeType2 PRIVATE
+		Common
+		GpCommon
+		"${FREETYPE_INCLUDE_DIR_ft2build}"
+		)
+
+	target_link_libraries(GpFontHandler_FreeType2 PRIVATE Freetype::Freetype)
+
+
+	add_executable(GenerateFonts EXCLUDE_FROM_ALL GenerateFonts/GenerateFonts.cpp AerofoilPortable/GpAllocator_C.cpp)
+	target_include_directories(GenerateFonts PRIVATE
+		Common
+		GpCommon
+		PortabilityLayer
+		AerofoilPortable
+		UnixCompat
+		)
+	target_link_libraries(GenerateFonts PortabilityLayer GpFontHandler_FreeType2)
+endif()
+
+add_executable(ConvertColorCursors EXCLUDE_FROM_ALL
+	ConvertColorCursors/ConvertColorCursors.cpp
+	AerofoilPortable/GpAllocator_C.cpp
+	stb/stb_image_write.c
+	)
+target_include_directories(ConvertColorCursors PRIVATE
+	Common
+	GpCommon
+	PortabilityLayer
+	AerofoilPortable
+	stb
+	UnixCompat
+	)
+target_link_libraries(ConvertColorCursors PortabilityLayer)
+
+
+file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/Packaged/Houses")
+
+set(DATA_FILES)
+
+function(add_data_file NAME)
+	list(APPEND DATA_FILES "Packaged/${NAME}")
+	set(DATA_FILES "${DATA_FILES}" PARENT_SCOPE)
+
+	cmake_parse_arguments(PARSE_ARGV 1 ARG
+		""
+		""
+		"BYPRODUCTS;COMMANDS"
+		)
+
+	if(ARG_BYPRODUCTS)
+		set(RM_BYPRODUCTS_COMMAND COMMAND "${CMAKE_COMMAND}" -E rm)
+		set(RM_BYPRODUCTS_PATHS "${ARG_BYPRODUCTS}")
+		list(TRANSFORM RM_BYPRODUCTS_PATHS PREPEND "${CMAKE_CURRENT_BINARY_DIR}/")
+	endif()
+
+	add_custom_command(
+		OUTPUT "Packaged/${NAME}"
+		BYPRODUCTS ${ARG_BYPRODUCTS}
+		${ARG_COMMANDS}
+		${RM_BYPRODUCTS_COMMAND} ${RM_BYPRODUCTS_PATHS}
+		VERBATIM
+		)
+endfunction()
+
+add_data_file(ApplicationResources.gpf
+	BYPRODUCTS
+		Packaged/ApplicationResources.gpr
+		Packaged/ApplicationResources.gpa
+	COMMANDS
+	DEPENDS MiniRez gpr2gpa FTagData MergeGPF
+	COMMAND MiniRez
+		"GliderProData/Glider PRO.r"
+		"${CMAKE_CURRENT_BINARY_DIR}/Packaged/ApplicationResources.gpr"
+	COMMAND gpr2gpa
+		"${CMAKE_CURRENT_BINARY_DIR}/Packaged/ApplicationResources.gpr"
+		DefaultTimestamp.timestamp
+		"${CMAKE_CURRENT_BINARY_DIR}/Packaged/ApplicationResources.gpa"
+		-patch ApplicationResourcePatches/manifest.json
+	COMMAND FTagData
+		DefaultTimestamp.timestamp
+		"${CMAKE_CURRENT_BINARY_DIR}/Packaged/ApplicationResources.gpf"
+		data ozm5 0 0 locked
+	COMMAND MergeGPF
+		"${CMAKE_CURRENT_BINARY_DIR}/Packaged/ApplicationResources.gpf"
+	WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
+	)
+
+add_data_file(Fonts.gpf
+	BYPRODUCTS
+		Packaged/Fonts.gpr
+		Packaged/Fonts.gpa
+		Packaged/FontCacheCatalog.bin
+		Packaged/FontCacheManifest.json
+		Packaged/CachedFont0.bin
+		Packaged/CachedFont1.bin
+		Packaged/CachedFont2.bin
+		Packaged/CachedFont3.bin
+		Packaged/CachedFont4.bin
+		Packaged/CachedFont5.bin
+		Packaged/CachedFont6.bin
+		Packaged/CachedFont7.bin
+		Packaged/CachedFont8.bin
+		Packaged/CachedFont9.bin
+		Packaged/CachedFont10.bin
+		Packaged/CachedFont11.bin
+		Packaged/CachedFont12.bin
+		Packaged/CachedFont13.bin
+		Packaged/CachedFont14.bin
```

Comment:
Is it possible to use a wildcard here or something that would not be dependent on there being a specific number of font presets?

Reply:
Globbing is a little tricky in CMake. There's `file(GLOB …)`, but that only runs when a CMake script (like `CMakeLists.txt`) runs, not at build time. I could have written a `cleanup.cmake` script and had that execute at build time via `cmake -P ${CMAKE_SOURCE_DIR}/cleanup.cmake`, but I decided on a better approach in ed4eb8e1f79b9f37739aae313baa408f898b5719. It helps that everything listed as a `BYPRODUCT` isn't needed except by `make clean`, so if you put these files in their own temporary directory, you can just recursively delete that directory afterwards.